### ISSUE TITLE
Shift-invert: error when not all eigenvalues have converged

### ISSUE THREAD

**Issue by nicolasbrughmans** [n-claes/legolas#122]
## Issue description

An error is thrown when not all eigenvalues have converged with shift-invert. It sometimes makes part of the spectrum unclickable or introduces a np.nan + 1j*0.0 solution at the location of the shift (like in this example).

## Bug report
**Minimal example for reproduction**

```
&gridlist
  gridpoints = 100
/

&equilibriumlist
  equilibrium_type = "RTI_theta_pinch"
/

&savelist
  basename_datfile = "EP22_fig_3a30"
/

&solvelist
  solver = "arnoldi"
  arpack_mode = "shift-invert"
  number_of_eigenvalues = 10
  which_eigenvalues = "LM"
  sigma = (0.49d0, 0.01d0)
/


```

**Actual result**

INFO    | solving eigenvalue problem...
WARNING | ARPACK failed to converge! (maxiter reached)
WARNING | number of iterations: 100
WARNING | number of converged eigenvalues: 2 / 10
...
/Users/nicolas/codes/legolas/post_processing/pylbo/utilities/datfile_utils.py:478: RuntimeWarning: invalid value encountered in greater
eigenvalues[np.where(np.absolute(eigenvalues) > 1e15)] = np.nan
...
/Users/nicolas/codes/legolas/post_processing/pylbo/visualisation/spectra/spectrum_single.py:52: RuntimeWarning: invalid value encountered in greater
(self._nonzero_w_idxs,) = np.where(np.abs(dataset.eigenvalues) > 1e-12)

**Version info**
- Legolas version: develop branch, b7dbd98f4a0ae6f5e5c050af3e52ff34d13b65cf

**Comment by n-claes** [comment]
Using the latest head on develop (1e178ba) I also get this warning about shift-invert not converging, but I'm unable to reproduce the error shown here. I did just merge a few PRs into the development branch though, including a datfile reader rework, so maybe it's fixed now?

**Comment by nicolasbrughmans** [comment]
Weirdly, it's not on my machine, but I have had to change zggev3 to zggev earlier due to some weirdness (see the Slack). Could it have to do with that? Apart from that, no differences with 1e178ba that I'm aware of...

**Comment by n-claes** [comment]
`zggev3` is the blocked variant of `zggev`, but has the same call signature and array returns so that shouldn't make any difference as far as I know. I could place an additional check when the datfile is read in, but I do want to know where the NaN comes from -- is this already within Legolas, or does it arise somehow during post-processing?

**Comment by n-claes** [comment]
It looks like this is fixed in the recent 2.0 release, as I'm unable to reproduce this. Closing the issue for now, feel free to reopen if it pops up again.
